pkg/client/clientcmd/client_config.go

```diff
@@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string, error) {
 	if len(configContext.Namespace) == 0 {
 		return api.NamespaceDefault, nil
 	}
+	if configContext.Namespace == "*" {
```

Comment:
AllNamespaces should be a flag on each command that needs to use it (not all commands need it) and it should be preserved as a boolean (all namespaces) rather than converting into api.NamespaceAll and passing it to code.

----- Original Message -----
> > @@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string,
> > error) {
> >  	if len(configContext.Namespace) == 0 {
> >  		return api.NamespaceDefault, nil
> >  	}
> > +	if configContext.Namespace == "*" {
> 
> I'm still not clear to handle this.  Is clearing namespace enough as
> commented above, or is there better one to pass all namespace as context?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7024/files#r29845278